src/Microsoft.OData.Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -671,6 +673,18 @@ private void MaterializeResolvedEntry(MaterializerEntry entry, bool includeLinks
             if (entity != null)
             {
                 entity.Context = this.EntityTrackingAdapter.Context;
+
+                if (!entry.IsTracking)
+                {
+                    int? streamDescriptorsCount = entry.EntityDescriptor.StreamDescriptors?.Count;
+                    IEdmEntityType entityType =
+                        this.EntityTrackingAdapter.Model.FindDeclaredType(entry.Entry.TypeName) as IEdmEntityType;
+                    if ( streamDescriptorsCount != null && (streamDescriptorsCount.Value > 0 ||
+                                                            ( entityType != null && entityType.HasStream)))

```

Comment:
```suggestion
                    if (streamDescriptorsCount.Value > 0 || entityType?.HasStream == true)
```
The spacings after the opening brackets would create warnings...
The checks for null ca be simplified where necessary.